integration/v3_lease_test.go

```diff
@@ -438,6 +440,8 @@ func TestV3LeaseFailover(t *testing.T) {
 		t.Fatal(lresp.Error)
 	}
 
+	expectedExp = time.Now().Add(time.Duration(lresp.TTL) * time.Second)
```

Comment:
lease proxy uses watch to detect leader loss where as the etcd server has some different logic for that. from my testing, lease proxy detects the leader loss right away. so at 
```go
func TestV3LeaseFailover(t *testing.T) {
...
	for {
               // first loop
		if err = lac.Send(lreq); err != nil {
			break
		}
		lkresp, rxerr := lac.Recv()
		if rxerr != nil {
                        // lease proxy returns leader loss at here.
			break
		}
                // lease proxy code never reaches here 
		expectedExp = time.Now().Add(time.Duration(lkresp.TTL) * time.Second)
		time.Sleep(time.Duration(lkresp.TTL/2) * time.Second)
	}
...

        // if expectedExp not set, waits forever
	time.Sleep(expectedExp.Sub(time.Now()) - 500*time.Millisecond)
}
```
the original code will not have `expectedExp` set , so the `time.Sleep(expectedExp.Sub(time.Now()) - 500*time.Millisecond)` sleeps forever.

in order to solve that issue, we need to set `expectedExp` at very beginning.

why did the etcd server code works?

from my investigation,

```go
func TestV3LeaseFailover(t *testing.T) {
...
	for {
                // loop one okay
                // loop two
		if err = lac.Send(lreq); err != nil {
			break
		}
		lkresp, rxerr := lac.Recv()
               // loop two encounters rxerr(request timeout)
		if rxerr != nil {
			break
		}
               // loop one sets expectedExp
		expectedExp = time.Now().Add(time.Duration(lkresp.TTL) * time.Second)
		time.Sleep(time.Duration(lkresp.TTL/2) * time.Second)
	}
...
// expectedExp is set at loop one, therefore the sleeps works.
time.Sleep(expectedExp.Sub(time.Now()) - 500*time.Millisecond)
...
}
```